### ISSUE TITLE
[Factions] Performance Issues

### ISSUE THREAD

**Issue by Wheatley1001** [Regalis11/Barotrauma#10493]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

**NOTE**: I am aware of the mega-ticket relating to performance issues. I am making this ticket separate because I did not want to muddy the ticket with faction related issues and stable issues. From a personal perspective, I've never had any performance issues with Barotrauma at any point in my time playing the game, so the crippling issues experienced in the faction branch are a cause for concern for me.

During the testing session, several players experienced game-ruining FPS issues. From input delays up to 10 seconds, to 4 FPS encounters, the session would occasionally be riddled with performance problems. Here are the notes gathered from these issues:
- **Spinelings** and **Tiger Threshers/Bone Threshers** cause FPS to tank
           - Heavy flooding exaggerates the issue
- Clicking on and off a periscope causes the game to drop to roughly 17 FPS for a brief moment
- Changing lighting settings does little to change the issue

Additionally, performance issues seemed to completely vanish once we entered the Great Sea. The biome before it was incredibly taxing on FPS rates, whilst the Great Sea was much easier for the game to handle. 

Below is a series of images captured by myself and other users in the session. I profusely apologise for the lack of the FPS counter in the top left, I didn't know it existed until much later on. I tried to captured the performance readings as soon as performance dipped to the lowest points. I hope they are helpful in finding some of the causes.

![image](https://user-images.githubusercontent.com/105495756/202923663-58004a59-bcaa-4ccb-8f60-de89ec9ee076.png)
![image](https://user-images.githubusercontent.com/105495756/202923667-eb348079-9252-4436-b1b4-c9c605e4d7e6.png)
![image](https://user-images.githubusercontent.com/105495756/202923670-c66bc17a-0b5b-427e-9256-ef8fd942ec8a.png)
![image](https://user-images.githubusercontent.com/105495756/202923675-5b37c423-aa49-4bd3-b874-63a48dc16695.png)
![image](https://user-images.githubusercontent.com/105495756/202923681-28483078-4a65-4406-940a-cc364fe69342.png)
![image](https://user-images.githubusercontent.com/105495756/202923687-9bd692d9-e0d2-4914-bc13-283775bdced3.png)

### Bug prevalence

Happens regularly

### Version

Faction/endgame test branch

### Which operating system did you encounter this bug on?

Windows

**Comment by Wheatley1001** [comment]
### PERFORMANCE UPDATE AS OF 100.6.0.0

NOTES: 
-Spinelings absolutely destroy performance. They, in no exaggerated terms, make the game completely unplayable. Constant 5 FPS and input lag of up to four seconds in some cases. Inputs would be held for longer than they were given, meaning players were swimming in directions they were actively telling the game to avoid.
-Threshers as a whole also absolutely tank performance. Anywhere between 6 - 17 FPS. Not as bad as Spinelings but still devastating.
-Water floods have remained impactful for performance issues. There have been no noticed changes in that area,

-Wrecks somehow provide high FPS rates of up to constant 100 FPS. This was in a Thalamus as well, although a dead one. Not entirely sure what's going on there.

### Performance screenshots from taken from tonight's session:

![image](https://user-images.githubusercontent.com/105495756/204066448-a995c904-cbcb-4739-afc2-88c89959b55b.png)
![image](https://user-images.githubusercontent.com/105495756/204066450-5504fbbb-55d6-4949-889a-59092721f23e.png)
![image](https://user-images.githubusercontent.com/105495756/204066612-0a0075ba-cc00-4edd-a35b-05206a09f6f7.png)
![image](https://user-images.githubusercontent.com/105495756/204066616-ff48a6ba-37f6-41c0-8a5d-54241b19ac32.png)
![image](https://user-images.githubusercontent.com/105495756/204066621-2169e159-2611-4593-bfc0-7b64e73b9d68.png)
![image](https://user-images.githubusercontent.com/105495756/204066630-51df7959-1272-439d-9e22-7de0bbe4de96.png)
![image](https://user-images.githubusercontent.com/105495756/204066636-6d4492ca-6d3f-4c9f-9dc1-adb117a55d6c.png)
![image](https://user-images.githubusercontent.com/105495756/204066644-559b258e-ecf2-4647-9f5e-f57767c0505c.png)
![image](https://user-images.githubusercontent.com/105495756/204067041-2c74d71e-9316-4697-afe4-67207928802d.png)
![image](https://user-images.githubusercontent.com/105495756/204067044-5cca83cd-9bdf-4100-b820-25219b532ff4.png)
![image](https://user-images.githubusercontent.com/105495756/204067168-b5c7f5a1-e459-49c9-9157-d4dea9b1e526.png)
![image](https://user-images.githubusercontent.com/105495756/204067172-f261fab7-a638-4e93-bd77-9b9094ceb63e.png)
![image](https://user-images.githubusercontent.com/105495756/204067176-dfbdc5b9-8e47-4a32-af75-b09fbd9aef63.png)
![image](https://user-images.githubusercontent.com/105495756/204067183-f1060439-dd62-4c75-800d-d030ae6d8827.png)
![image](https://user-images.githubusercontent.com/105495756/204067178-8eeaf470-98cc-4590-8e0f-91ac9b9b0ae5.png)
![image](https://user-images.githubusercontent.com/105495756/204067199-e92396e0-a0da-4049-80ce-c0481b71afdc.png)

From this point onward, screenshots only show FPS as Rokvach informed us that showperf lowered performance.

![image](https://user-images.githubusercontent.com/105495756/204067298-a9a53ad8-3a17-4056-8346-0cdff9e1984e.png)
![image](https://user-images.githubusercontent.com/105495756/204067330-e8d4f427-fbaa-4993-b6c0-c49b703d5bd4.png)
![image](https://user-images.githubusercontent.com/105495756/204067337-e607040b-7690-42fc-8dbf-4a4ec3ce32a3.png)
![image](https://user-images.githubusercontent.com/105495756/204067410-4b70232c-17f4-4b9b-ada2-665dae2e1694.png)
![image](https://user-images.githubusercontent.com/105495756/204067414-5fc2b97c-b73c-4ad2-a371-72f7326db0ed.png)
![image](https://user-images.githubusercontent.com/105495756/204067485-dfef512c-fc7e-48b6-92a3-acad4d5dc0fa.png)
![image](https://user-images.githubusercontent.com/105495756/204067581-30a8e1e8-af6e-46b5-8ba9-a31bc5720ae1.png)
![image](https://user-images.githubusercontent.com/105495756/204067600-1a34495f-972a-4bdd-9c88-da5e26cc1577.png)
![image](https://user-images.githubusercontent.com/105495756/204067603-c65e68b8-6fbb-48b3-8ad5-9f38363914ea.png)

**Comment by Regalis11** [comment]
Thank you for the report and the pics!

To me it seems like this is the same kind of "mystery FPS drop" with no clear cause that we've been trying to diagnose for a long time. I wrote a bit more about that [here](https://github.com/Regalis11/Barotrauma/issues/9396#issuecomment-1160115410). Basically, according to the performance statistics in your pics, there doesn't seem to be any clear cause for such a low framerate. I also gave this a quick test myself, testing the campaign map and spawning a dozen or so spinelings and threshers around the sub, and the FPS stayed around 120-144. Need to do more testing, could be that a large number of players for example makes the issues worse.
